sunpy/instr/aia/response.py

```diff
@@ -43,267 +44,48 @@ class description here!
-        self.primary = kwargs.get('primary', var['primary'])
-        self.secondary = kwargs.get('secondary', var['secondary'])
-        self.ccd = kwargs.get('ccd', var['ccd'])
-        self.instrument_contamination = kwargs.get('contam', var['contam'])  # nan in 1600 and 1700
-        # self.cross_area = kwargs.get('cross_area', var['cross_area'][0])
-        # 1600, 1700 channels have arrays for some attributes #TODO: figure out why/ which is best
-        if type(self.primary) == np.ndarray: # previously self.primary[0]
-            self.primary = self.primary[0]
-            self.secondary = self.secondary[0]
-
-        # eff_area is not in all channels
-        try:
-            self.eff_area = kwargs.get('eff_area', var['effarea'])
-        except KeyError:
-            print(wavelength, 'has no eff_area keyword.')
-            pass
-
-    def aia_inst_genx_to_dataframe(self, input_directory, version):
-        """
-
-            Give the directory to a .genx directory to recursive search inside and return the the EUV and UV instrument files.
-            output = list of paths to instrument files
-
-        Parameters
-        ----------
-        path_to_file : string, the path location that leads to ssw_aia_response_data/aia_V6_all_fullinst.
-
-        :param save: bool, if True, will save a dataframe containing all of the extracted instrument data to
-
-        """
-        check = []  # make sure not duplicated
-        file_paths = []
-
-        for root, dirs, files in os.walk(input_directory, topdown=False):
-
-            for name in files:
-                if (os.path.join(root, name)) not in check and name.endswith('_fullinst') and str(version) in str(name):
-                    check.append(os.path.join(root, name))
-                    file_paths.append(os.path.join(root, name))
-
-        assert len(file_paths) != 0, 'Did not find aia_' + str(version) + '_fullinst files in directory.'
-
-        # fill dataframe with desired properties and save csv file
-        self.dataframe = self.save_properties_to_dataframe(file_paths, version)
-
-    def save_properties_to_dataframe(self, files, version):
-        """
-        This definition reads the instrument file aia_V6_all_fullinst, which was obtained from ssw_aia_response_data inside
-        ssw_aia_response_genx.tar.gz (an output file saved from SolarSoft Ware (SSW)). It will extract the instrument data and save
-        it a dataframe file for easier access.
-
-
-        Returns
-        -------
-        :returns self.dataframe: dataframe, stores the dictionary for easy access. Each column lists the channel information and each row is a different peroperty from aia_inst_genx.
-
-        :returns 'channel_properties.csv', dataframe with wavelength centers for each channel is a column and listed in the column are the properties from the is returned where it has keys of the properties of the channel
+    def __init__(self, dataframe, channel = 'all', **kwargs): #data_table,
```

Comment:
I'm a little confused as to what is going on here. Is the user supposed to read in the dataframe from `read_genx` before instantiating `Response`?

I think the better way to do this is just to have `__init__` (or a new `read_data` method) call `read_genx`. This combines two steps into one and this way the interface is the same regardless of where the data is being read from.